javalib/src/main/scala/java/lang/System.scala

```diff
@@ -383,8 +385,8 @@ private final class JSConsoleBasedPrintStream(isErr: scala.Boolean)
   override def close(): Unit = ()
 
   private def doWriteLine(line: String): Unit = {
-    if (LinkingInfo.targetPureWasm) { // isWASI
-      // TODO
+    if (LinkingInfo.targetPureWasm) {
+      WasmSystem.print(line + "\n")
```

Comment:
Consider moving the `+ "\n"` to the component model-specific code. The `console.log`-based implementation of the `print()` magic helper adds a newline in a way that cannot be avoided.
```suggestion
      WasmSystem.print(line)
```